### ISSUE TITLE
✨ feat: Selection of models for system optimization

### ISSUE THREAD

**Issue by Daehyun-Bigbread** [Kakaotech-18-Ecommerce/Kakaotech-18-AI#59]
# Title

- 추천 시스템 최적화를 위한 모델 선정

- 모델 최적화 프로세스:
1. 다양한 모델을 훈련시킨 후, 각 모델의 MSE, MAE, RMSE, R2R^2R2, MAPE 값을 비교합니다.
2. 가장 낮은 MSE, MAE, RMSE 값을 가지는 모델과 가장 높은 R2R^2R2 값을 가지는 모델을 선택합니다.
3. 특정 지표에서 성능이 더 좋더라도, 모든 지표를 고려하여 종합적으로 성능이 우수한 모델을 최종적으로 선택합니다.

Q. 왜 이러한 많은 모델들을 Research 하는데 사용?
-  각 모델이 고유한 장점과 특징을 가지고 있기 때문. 다양한 회귀 및 앙상블 모델을 사용하는 이유는 데이터의 특성과 문제의 성격에 따라 특정 모델이 더 나은 성능을 발휘할 수 있기 때문.

# TODO (모델 Research)

- [ ] Linear Regression
- [ ] Ridge Regression
- [ ] Lasso Regression
- [ ] Random Forest
- [ ] XGBoost
- [ ] LightGBM
- [ ] Gradient Boosting
- [ ] Support Vector Machine (SVM)
- [ ] K-Nearest Neighbors (KNN)
- [ ] Ensemble Techniques

# etc

- 각 모델마다 평가지표를 MSE, MAE, RMSE, R2R^2R2, MAPE 값을 비교해야합니다.
- 가장 낮은 MSE, MAE, RMSE 값을 가지는 모델과 가장 높은 R2R^2R2 값을 가지는 모델을 선택합니다.
